wsr-tic-tac-toe/src/app.tsx

```diff
@@ -1,71 +1,39 @@
 import React, { useState, useEffect } from 'react';
 import { st, classes } from './app.st.css';
-import { Button, Modal, MessageBoxFunctionalLayout } from 'wix-style-react';
-import { InfoIcon } from 'wix-style-react';
-import {
-    CREATE_GAME_BOARD,
-    SET_GAME_BOARD_SIZE,
-    BOARD_SIZE,
-    SET_BOARD_SIZE_DESCRIPTION,
-    CONFIRM_TEXT,
-    SET_BOARD_SIZE_INFO_TOOLTIP,
-} from './constants';
-import { InputWithLabel } from 'wix-style-react';
-import {
-    createBoard,
-    generateWinningConditions,
-    removeBoard,
-    setGameState,
-} from './redux/actions';
+import { Button } from 'wix-style-react';
+import { CREATE_GAME_BOARD } from './constants';
+import { setGameState } from './redux/actions';
 import { useDispatch, useSelector } from 'react-redux';
 import { GameBoard } from './components/game-board';
 import { AppState, SquareState, GAME_STATE } from './redux/types';
+import { CreateGameModal } from './components/create-game-modal';
+import { checkIfPlayerWon } from './utils/game-logic';
 
 export interface AppProps {
     className?: string;
 }
 
 export const App: React.FC<AppProps> = ({ className }) => {
     const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
```

Comment:
modal visibility is something that should be stored in redux. It's not just visual, it's the state of your app. Makes including modal easier too, since it can handle it's own visibility via redux, instead of having some parent component state

Reply:
If we had more modals in the application, would we have a reducer for each of them ?

Reply:
Depends on how you want it to behave.
Eg, you can have one modal reducer, which identifies modal that is open by some ID. {isOpen: bool, id: string}
If you want possibility of multiple modals being open at once, then I'd say a modals reducer with IDs as keys {isModalXOpen: bool, isModalYOpen: bool, ...}.

In any case you need a way to ask the redux state if a specific modal is open anyway. Some selector like isMyPreciousModalOpen(state) or isModalOpen(state, ID_OF_MODAL)

Reply:
Yes, what I meant was "state" not "reducer" but you answered my question :)